code/modules/status_system/statusEffects.dm

```diff
@@ -345,209 +345,103 @@
-				if(60 to INFINITY)
-					. = 5
-
-		modify_change(change)
-			. = change
-
-			if (owner && ismob(owner) && change > 0)
-				var/mob/M = owner
-				SEND_SIGNAL(M, COMSIG_MOB_GEIGER_TICK, get_stage(duration + change))
-				var/percent_protection = clamp(GET_ATOM_PROPERTY(M, PROP_MOB_RADPROT), 0, 100)
-				percent_protection = 1 - (percent_protection/100) //scale from 0 to 1
-				. *= percent_protection
-
-		proc/update_lights(add = 1)
-			owner.remove_simple_light("neutron_rad")
-			owner.remove_medium_light("neutron_rad")
-			if(add)
-				if(stage < 4)
-					owner.add_simple_light("neutron_rad", list(4, 62, 155, stage * 50))
-				else
-					owner.add_medium_light("neutron_rad", list(4, 62, 155, stage * 50))
-
-		onAdd(optional=null)
-			. = ..()
-			update_lights()
-
-		onRemove()
-			update_lights(0)
-			. = ..()
-
-		onChange(optional=null)
-			. = ..()
-			update_lights()
+			. = 0
+			switch(val) //0.4 Sv is radiation poisoning, 2 Sv is fatal in some cases, 4 Sv is fatal without treatment
+				if(0 to 0.35)
+					. = 0
+				if(0.35 to 0.6)
+					. = 1 //you might feel sick
+				if(0.6 to 1.2)
+					. = 2 //you're getting into dangerous teritory
+				if(1.2 to 2)
+					. = 3 //you're at a 50/50 of kicking it
+				if(2 to 3)
+					. = 4 //more like 70/30 now
+				if(3 to 4)
+					. = 5 //you will die without treatment
+				if(4 to INFINITY)
+					. = 6 //you will die.
 
 		onUpdate(timePassed)
-
 			var/mob/M = null
 			if(ismob(owner))
 				M = owner
+			else
+				return ..(timePassed)
 
-
-			damage_tox = (sqrt(min(duration, 2 MINUTES)/20 + 5) - 0.5)
-			damage_brute = damage_tox/2
-
-			stage = get_stage(duration)
+			stage = get_stage(M.radiation_dose)
 			switch(stage)
-				if(1)
+				if(0)
 					howMuch = ""
+					damage_tox = 0
+					damage_burn = 0
+				if(1)
+					howMuch = "barely " //you'll be fine
 				if(2)
-					howMuch = "significantly "
+					howMuch = "slightly " //you don't feel so good
 				if(3)
-					howMuch = "very much "
+					howMuch = "moderately " // not great, not terrible
 				if(4)
-					howMuch = "extremely "
+					howMuch = "extremely " //oh no, you're very sick
 				if(5)
-					howMuch = "horribly "
+					howMuch = "fatally " //congrats, you're dead in a minute
+				if(6)
+					howMuch = "totally " // you are literally dying in seconds
+			if(stage > 0)
+				visible = TRUE
+				var/damage_total = 5 * (M.radiation_dose**1.4 - tanh(M.radiation_dose**1.6))
+				damage_tox = prob(70) * damage_total
+				damage_burn = prob(30) * damage_total
+			else
+				visible = FALSE
 
-			if(M)
-				SEND_SIGNAL(M, COMSIG_MOB_GEIGER_TICK, stage)
-				if(locate(/obj/item/implant/health) in M)
-					src.visible = 1
-				else
-					src.visible = 0
-				if (prob(stage - !!(M.traitHolder?.hasTrait("stablegenes"))) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
+
+			if(stage > 0 && !isdead(M))
+				if (prob(max(stage-(2+M.traitHolder?.hasTrait("stablegenes")),0)**2) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
```

Comment:
should have a cooldown on the check here too probably

Reply:
and maybe actually bump the stage offset down by 1. will probably need some tweaking to get this feeling right